### PR TITLE
Renaming to "AllocatorRecipeResolver"

### PR DESCRIPTION
**Pull request by alxrsngrtn**

### CODE REVIEW

**Comment by piotrswigon** [comment]
I am not sure this is the right rename, let me explain:

In Arcs a `Plan` is a `Recipe` that is resolved. What does it mean to be resolved is described in some detail here: https://docs.google.com/document/d/1OuPZYk6q-l0yxRiWzrE2H37vtRJQ0Qx6FhIkddhl_1c/edit#bookmark=id.ba1bg0fo72iv

We already have a `RecipeResolver`. Which, as its name suggests, is resolving recipes, i.e. turning a `Recipe` into a `Plan`.

What does a `PlanResolver` therefore do? It cannot resolve a Plan, as plan by definition is resolved.

What the class in question does is preparing a recipe for deployment on the ProdEx architecture. It turns out that some details of that currently different than instantiating a recipe in TypeScript runtime, so this class deals with the needs of ProdEx explicitly.

Ideally the ArcHost based architecture will be adopted by the TS codebase, and then maybe we will be able to unify the resolution steps, but today it's not yet the case.

**Comment by alxrsngrtn** [comment]
@piotrswigon What do you think about this rename (allocator-resolver)?

**Comment by piotrswigon** [comment]
:+1:

File: src/tools/allocator-resolver.ts [hunk 1]
```diff
@@ -23,17 +23,17 @@ import {DatabaseStorageKey} from '../runtime/storageNG/database-storage-key.js';
...
-    this.name = 'StorageKeyRecipeResolverError';
+    this.name = 'AllocatorResolverError';
   }
 }
 
 /**
  * Responsible for resolving recipes with storage keys.
  */
-export class StorageKeyRecipeResolver {
+export class AllocatorResolver {
```
Comment (piotrswigon) [inline src/tools/allocator-resolver.ts:36]: I would prefer "AllocatorRecipeResolver" TBH + a new comment explaining the relationship of this class to the Allocator / distributed architecture.

Comment (alxrsngrtn) [reply src/tools/allocator-resolver.ts:36]: I'm happy with that. I'll make the change.

**Comment by alxrsngrtn** [comment]
Will merge after TAP is fixed and presubmit passes.

**Pull request merged**